### PR TITLE
Integration powsybl-4.0.0-RC2

### PR DESCRIPTION
**Pull request by annetill**
Signed-off-by: Anne Tilloy <anne.tilloy@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

- Use `List<Contingency>` instead of `ContingencyProvider` in sensitivity-analysis.

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by floriand-e2r — APPROVED** [review]
You could maybe give an empty list (and then testing if it's empty) instead of a null list

File: pom.xml [hunk 1]
```diff
@@ -229,6 +229,12 @@
             <version>${powsyblcore.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (annetill) [inline pom.xml:232]: @floriand-e2r Can you please check this. I am not sure about my modification.

Comment (floriand-e2r) [reply pom.xml:232]: I'd rather use Supplier<T> (instead of Provider<T>) to avoid adding this dependency

Comment (annetill) [reply pom.xml:232]: Do I have to make the code modification?

**Pull request closed without merging**